### PR TITLE
[21825] Fix generated code compilation when a union uses an alias with scope

### PR DESCRIPTION
**Pull request by richiware**
## Description

When a union uses an alias with scope, the generated source code fails compiling. This PR fixes it.

Depends on:

- eprosima/dds-types-test#39
- eprosima/idl-parser#164

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/main/java/com/eprosima/fastcdr/idl/templates/TypesHeader.stg [hunk 1]
```diff
@@ -836,7 +836,15 @@ $if(member.typecode.primitive)$
...
 $else$
-member_destructor_ = [&]() {$union_member_destroy_call(member)$\};
+member_destructor_ = [&]()
+{
+    $if(member.typecode.isAliasType)$
+    $if(member.typecode.hasScope)$
+    using namespace $member.typecode.scope$;
+    $endif$
+    $endif$
+    $union_member_destroy_call(member)$\
```
Comment (MiguelCompany) [inline src/main/java/com/eprosima/fastcdr/idl/templates/TypesHeader.stg:846]:
```suggestion
    $union_member_destroy_call(member)$
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/fastcdr/idl/templates/TypesHeader.stg (hunk 1 above)
Comment (richiware) [reply src/main/java/com/eprosima/fastcdr/idl/templates/TypesHeader.stg]: Done in 009bc8d

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**